app/models/concerns/fiscal_year_mixin.rb

```diff
@@ -9,5 +11,11 @@ def self.which_fiscal_year(year, month)
         year
       end
     end
+
+    def self.for_fiscal_year(year)
```

Comment:
Since we use this mixin in one non-AR class, scope did not work because
that is part of AR (returns "undefined method scope").

 I did have a weird feeling about this mixin change because the mixin is
now used 3 times: twice in AR classes and once in a regular Ruby class. And
in each instance only one method or the other is used. But I did like the
idea of having these methods together since they are logically related.

Ideas on improvement?

On Wednesday, December 16, 2015, Peter Karman <notifications@github.com>
wrote:

> In app/models/concerns/fiscal_year_mixin.rb
> <https://github.com/18F/C2/pull/876#discussion_r47789680>:
>
> > @@ -9,5 +11,11 @@ def self.which_fiscal_year(year, month)
> >          year
> >        end
> >      end
> > +
> > +    def self.for_fiscal_year(year)
>
> This was a scope and now a class method? Are they, strictly speaking, the
> same thing?
>
>
> http://blog.plataformatec.com.br/2013/02/active-record-scopes-vs-class-methods/
> was illuminating to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/C2/pull/876/files#r47789680>.
>